core/modules/utils/dom/http.js

```diff
@@ -13,11 +13,201 @@ Browser HTTP support
+	this.completionActions = options.oncompletion;
+	this.progressActions = options.onprogress;
+	this.bindStatus = options["bind-status"];
+	this.bindProgress = options["bind-progress"];
+	this.method = options.method || "GET";
+	this.body = options.body || "";
+	this.variables = options.variables;
+	var url = options.url;
+	$tw.utils.each(options.queryStrings,function(value,name) {
+		url = $tw.utils.setQueryStringParameter(url,name,value);
+	});
+	$tw.utils.each(options.passwordQueryStrings,function(value,name) {
+		url = $tw.utils.setQueryStringParameter(url,name,$tw.utils.getPassword(value) || "");
+	});
+	this.url = url;
+	this.requestHeaders = {};
+	$tw.utils.each(options.headers,function(value,name) {
+		self.requestHeaders[name] = value;
+	});
+	$tw.utils.each(options.passwordHeaders,function(value,name) {
+		self.requestHeaders[name] = $tw.utils.getPassword(value) || "";
+	});
+}
+
+HttpClientRequest.prototype.send = function(callback) {
+	var self = this,
+		setBinding = function(title,text) {
+			if(title) {
+				this.wiki.addTiddler(new $tw.Tiddler({title: title, text: text}));
+			}
+		};
+	if(this.url) {
+		setBinding(this.bindStatus,"pending");
+		setBinding(this.bindProgress,"0");
+		// Set the request tracker tiddler
+		var requestTrackerTitle = this.wiki.generateNewTitle("$:/temp/HttpRequest");
+		this.wiki.addTiddler({
+			title: requestTrackerTitle,
+			tags: "$:/tags/HttpRequest",
+			text: JSON.stringify({
+				url: this.url,
+				type: this.method,
+				status: "inprogress",
+				headers: this.requestHeaders,
+				data: this.body
+			})
+		});
+		this.xhr = $tw.utils.httpRequest({
+			url: this.url,
+			type: this.method,
+			headers: this.requestHeaders,
+			data: this.body,
+			callback: function(err,data,xhr) {
+				var hasSucceeded = xhr.status >= 200 && xhr.status < 300,
+					completionCode = hasSucceeded ? "complete" : "error",
+					headers = {};
+				$tw.utils.each(xhr.getAllResponseHeaders().split("\r\n"),function(line) {
+					var pos = line.indexOf(":");
+					if(pos !== -1) {
+						headers[line.substr(0,pos)] = line.substr(pos + 1).trim();
+					}
+				});
+				setBinding(self.bindStatus,completionCode);
+				setBinding(self.bindProgress,"100");
+				var resultVariables = {
+					status: xhr.status.toString(),
+					statusText: xhr.statusText,
+					error: (err || "").toString(),
+					data: (data || "").toString(),
+					headers: JSON.stringify(headers)
+				};
+				self.wiki.addTiddler(new $tw.Tiddler(self.wiki.getTiddler(requestTrackerTitle),{
+					status: completionCode,
+				}));
+				self.wiki.invokeActionString(self.completionActions,undefined,$tw.utils.extend({},self.variables,resultVariables),{parentWidget: $tw.rootWidget});
+				callback(hasSucceeded ? null : xhr.statusText);
+				// console.log("Back!",err,data,xhr);
+			},
+			progress: function(lengthComputable,loaded,total) {
+				if(lengthComputable) {
+					setBinding(bindProgress,"" + Math.floor((loaded/total) * 100))
```

Comment:
The zotero pos252 example is giving me `Uncaught ReferenceError: bindProgress is not defined`. Maybe you mean this?

```suggestion
					setBinding(self.bindProgress,"" + Math.floor((loaded/total) * 100))
```